spark2/src/main/java/org/apache/iceberg/spark/source/Reader.java

```diff
@@ -329,8 +335,10 @@ public boolean enableBatchRead() {
 
       boolean batchReadsEnabled = batchReadsEnabled(allParquetFileScanTasks, allOrcFileScanTasks);
 
+      boolean hasTimestampWithoutZone = SparkUtil.hasTimestampWithoutZone(lazySchema());
+
       this.readUsingBatch = batchReadsEnabled && hasNoDeleteFiles && (allOrcFileScanTasks ||
-          (allParquetFileScanTasks && atLeastOneColumn && onlyPrimitives));
+              (allParquetFileScanTasks && atLeastOneColumn && onlyPrimitives && !hasTimestampWithoutZone));
```

Comment:
Why do we need the check here on the batch reading code?

Reply:
this code was added by @bkahloon 
From my point of view we don't need this check